diff --git a/node_modules/@real-token/aa-core/dist/lib.cjs/provider.js b/node_modules/@real-token/aa-core/dist/lib.cjs/provider.js
index 886f463..9b9f7e8 100644
--- a/node_modules/@real-token/aa-core/dist/lib.cjs/provider.js
+++ b/node_modules/@real-token/aa-core/dist/lib.cjs/provider.js
@@ -504,7 +504,7 @@ const AAProvider = ({
    * await addTransaction({ to: "0x...", value: 1000000000000000000n });
    */
   const addTransaction = react$1.useCallback(tx => {
-    if (!connectors[0]) throw new Error("User is not connected");
+    if (!connectors[0] && !connections[0]?.connector) throw new Error("User is not connected");
     const txs = getState().txs;
     if (s.aaSDK && !s.walletAddress) throw new Error("wallet address is undefined");
     const key = s.aaSDK ? index.TxType.AA : index.TxType.EOA;
@@ -522,7 +522,7 @@ const AAProvider = ({
       type: key,
       index: newTxs[key].length - 1
     };
-  }, [connectors, getState, s.aaSDK, s.walletAddress]);
+  }, [connectors, connections, getState, s.aaSDK, s.walletAddress]);
   /**
    * Sends all bundled transactions in the cart.
    * @param {EstimateProps} [p] - Optional gas and paymaster details.
@@ -789,7 +789,7 @@ const AAProvider = ({
    * });
    */
   const addSignature = react$1.useCallback((signatureType, data) => {
-    if (!connectors[0]) throw new Error("User is not connected");
+    if (!connectors[0] && !connections[0]?.connector) throw new Error("User is not connected");
     const txs = getState().txs;
     if (s.aaSDK && !s.walletAddress) throw new Error("wallet address is undefined");
     const key = s.aaSDK ? index.TxType.AA_SIGNATURES : index.TxType.EOA_SIGNATURES;
diff --git a/node_modules/@real-token/aa-core/dist/lib.esm/provider.js b/node_modules/@real-token/aa-core/dist/lib.esm/provider.js
index 4f85933..e9f481b 100644
--- a/node_modules/@real-token/aa-core/dist/lib.esm/provider.js
+++ b/node_modules/@real-token/aa-core/dist/lib.esm/provider.js
@@ -513,7 +513,7 @@ const AAProvider = ({
    * await addTransaction({ to: "0x...", value: 1000000000000000000n });
    */
   const addTransaction = useCallback(tx => {
-    if (!connectors[0]) throw new Error("User is not connected");
+    if (!connectors[0] && !connections[0]?.connector) throw new Error("User is not connected");
     const txs = getState().txs;
     if (s.aaSDK && !s.walletAddress) throw new Error("wallet address is undefined");
     const key = s.aaSDK ? TxType.AA : TxType.EOA;
@@ -531,7 +531,7 @@ const AAProvider = ({
       type: key,
       index: newTxs[key].length - 1
     };
-  }, [connectors, getState, s.aaSDK, s.walletAddress]);
+  }, [connectors, connections, getState, s.aaSDK, s.walletAddress]);
 
   /**
    * Sends all bundled transactions in the cart.
@@ -810,7 +810,7 @@ const AAProvider = ({
    * });
    */
   const addSignature = useCallback((signatureType, data) => {
-    if (!connectors[0]) throw new Error("User is not connected");
+    if (!connectors[0] && !connections[0]?.connector) throw new Error("User is not connected");
     const txs = getState().txs;
     if (s.aaSDK && !s.walletAddress) throw new Error("wallet address is undefined");
     const key = s.aaSDK ? TxType.AA_SIGNATURES : TxType.EOA_SIGNATURES;
